### ISSUE TITLE
Broken multi progress bar

### ISSUE THREAD

**Issue by eknauer** [piotrmurach/tty-progressbar#28]
If the width option is specified for a child bar and this option is smaller than the total option of this child bar, the main bar breaks in two lines.

For example run this code:

```ruby
#!/usr/bin/env ruby

require 'tty-progressbar'

bars = TTY::ProgressBar::Multi.new("main [:bar] :percent")

bar1 = bars.register "foo [:bar] :percent", total: 50, width: 20
bar2 = bars.register "bar [:bar] :percent", total: 200, width: 20
bar3 = bars.register "baz [:bar] :percent", total: 100, width: 20

th1 = Thread.new { 50.times { sleep(0.05); bar1.advance } }
th2 = Thread.new { 200.times { sleep(0.05); bar2.advance } }
th3 = Thread.new { 100.times { sleep(0.02); bar3.advance } }

[th1, th2, th3].each(&:join)
```

There is a workaround:

```ruby
#!/usr/bin/env ruby

require 'tty-progressbar'

module TTY
  class ProgressBar
    class Multi
      def register(format, options = {})
        observable = options.delete(:observable) { true }
        bar = TTY::ProgressBar.new(format, @options.merge(options))

        synchronize do
          bar.attach_to(self)
          @bars << bar
          observe(bar) if observable
          if @top_bar
            width = total
            width = options[:width] if options[:width] 
            @top_bar.update(total: total, width: width)
          end
        end

        bar
      end
    end
  end 
end

bars = TTY::ProgressBar::Multi.new("main [:bar] :percent")

bar1 = bars.register "foo [:bar] :percent", total: 50, width: 20
bar2 = bars.register "bar [:bar] :percent", total: 200, width: 20
bar3 = bars.register "baz [:bar] :percent", total: 100, width: 20

th1 = Thread.new { 50.times { sleep(0.05); bar1.advance } }
th2 = Thread.new { 200.times { sleep(0.05); bar2.advance } }
th3 = Thread.new { 100.times { sleep(0.02); bar3.advance } }

[th1, th2, th3].each(&:join)
```
Tested on  ubuntu 16.04 ruby version 2.3.1p112

**Comment by piotrmurach** [comment]
Thanks for using this library and reporting this issue!

You workaround though it may work doesn't solve the issue. If you happen to pass a width exceeding terminal width you will have the same problem. The issue is that by default multibar 'figures out' the width to match the sum of all totals - this is clearly wrong for large totals! There needs to be a check to see if total exceeds terminal columns size. 

Also, if each registered bar has its own width then this should take precedence over the 'automatic' width assigned to the multibar. Given your example, I would expect the multi bar to be `60` columns wide unless it exceeds the terminal window size, in other words, the width to be cumulative.

If you have time to work on PR, the I would suggest to add test `width_spec.rb` to [spec/unit/multi](https://github.com/piotrmurach/tty-progressbar/tree/master/spec/unit/multi) directory.

**Comment by piotrmurach** [comment]
Released `v0.15.0` that fixes this issue!

**Comment by eknauer** [comment]
Thank you!
